dspace-api/src/main/java/org/dspace/eperson/GroupServiceImpl.java

```diff
@@ -158,35 +158,32 @@ public boolean isMember(Group owningGroup, Group childGroup) {
 
     @Override
     public boolean isMember(Context context, Group group) throws SQLException {
-        return isMember(context, group.getName());
+        return isMember(context, context.getCurrentUser(), group);
     }
 
     @Override
-    public boolean isMember(final Context context, final String groupName) throws SQLException {
-        // special, everyone is member of group 0 (anonymous)
-        if (StringUtils.equals(groupName, Group.ANONYMOUS))
-        {
-            return true;
-        } else if(context.getCurrentUser() != null) {
-            EPerson currentUser = context.getCurrentUser();
+    public boolean isMember(Context context, String groupName) throws SQLException {
+        return isMember(context, context.getCurrentUser(), findByName(context, groupName));
+    }
 
-            //First check the special groups
-            List<Group> specialGroups = context.getSpecialGroups();
-            if(CollectionUtils.isNotEmpty(specialGroups)) {
-                for (Group specialGroup : specialGroups)
-                {
-                    //Check if the current special group is the one we are looking for OR retrieve all groups & make a check here.
-                    if(StringUtils.equals(specialGroup.getName(), groupName) || allMemberGroups(context, currentUser).contains(findByName(context, groupName)))
-                    {
-                        return true;
-                    }
-                }
-            }
-            //lookup eperson in normal groups and subgroups
-            return epersonInGroup(context, groupName, currentUser);
```

Comment:
I would leave the `epersonInGroup` method in place as this was written especially to make the `isMember` method more performant (see ticket https://jira.duraspace.org/browse/DS-3004): Saving the retrieval of the Group object from the database and checking membership with 1 query so that it's easily cacheable by Hibernate.

I think it is too expensive to retrieve all the groups of an eperson (multiple times) for each web request if not absolutely necessary (read: when there are no special groups).